src/config/preferences.c

```diff
@@ -2323,6 +2326,8 @@ _get_default_string(preference_t pref)
         return "xdg-open";
     case PREF_URL_OPEN_CMD:
         return "xdg-open %u";
+    case PREF_COMPOSE_EDITOR:
+        return "/usr/bin/vim";
```

Comment:
May be we should consider using `/usr/bin/vi` as default. That should exist on almost any *nix machine. For example OpenBSD installs to `/usr/local/bin` but there is a basic `vi` in `/usr/bin`.

Reply:
does `$EDITOR` work?

Reply:
Maybe also just use `xdg-open` I think it should work since you use `.txt` extension, right? Then its similar to other executables and user can set to their fav editor in `/executable`.

Reply:
> Maybe also just use `xdg-open` I think it should work since you use `.txt` extension, right? Then its similar to other executables and user can set to their fav editor in `/executable`.

Maybe we can recommend it. But I think vi will be on all systems. xdg-open is maybe not installed in headless systems.

Reply:
true. also right now it doesnt seem to work with graphical editors (did a quick test with gedit only)

Reply:
Why not "vim" instead of "/usr/bin/vim"? This breaks on freebsd, GuixSD and NixOS because vim is not in `/usr/bin/vim`.
Also users are no longer able to have self-compiled vims.

Reply:
> Why not "vim" instead of "/usr/bin/vim"? This breaks on freebsd, GuixSD and NixOS because vim is not in `/usr/bin/vim`.
> Also users are no longer able to have self-compiled vims.

User can set the editor via `~/.config/profanity/profrc`.

```
[ui]
compose.editor=/path/editor
```

Reply:
Sure but what is the advantage of the absolute path?

Reply:
It doesn't "break" on freebsd etc. You just need to configure it ;). We need absolute path because of the function we use to call it.